src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -91,16 +107,48 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
                             },
                         });
                     })
-                    .then(nonce =>
-                        this._store.dispatch(this._paymentActionCreator.submitPayment({
-                            methodId: payment.methodId,
-                            paymentData: { nonce },
-                        }))
+                    .then(nonce => {
+                            if (nonce !== undefined) {
+                                return this._store.dispatch(this._paymentActionCreator.submitPayment({
+                                    methodId: payment.methodId,
+                                    paymentData: { nonce },
+                                }));
+                            } else {
+                                this._goToOrderConfirmation();
+
+                                return this._store.getState();
+                            }
+                        }
                     ));
             });
     }
 
     finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
+
+    private _prepareForReferredRegistration(checkoutId: string): Promise<Response> {
+        const url = '/internalapi/v1/payments/zip';
+        const options = {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            },
+            body: toFormUrlEncoded({
```

Comment:
As I said above, you shouldn't need to encode this if you're sending the request body as JSON.

Reply:
Hi @davidchin , I tried to do this, but I was unable to access `externalId` in the post params, can you give some pointers on how to send this request correctly without using `toFormUrlEncoded `?